opendistro-for-elasticsearch/config/hook-helper.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+create_credentials_delay() {
+    CURL_STATUS="1"
+    BREAKER=0
+    while [ "$CURL_STATUS" != "0" ]; do
+    sleep 5
```

Comment:
I saw the email, but for the life of me I can't find the comment.  

The reason we can't allow sleep in hooks (and that applies to here as well since it's called from a hook) is that all lifecycle hooks (with the exception of run) run in the same thread as the supervisor. That means that while this is executing, the supervisor is not able to do other work. If curl status were to never return 0, then the supervisor is effectively deadlocked. 

There's work in flight to run each hook in it's own thread, and I believe one or two of them in addition to run already do, but we still shouldn't allow it in core until all that work is complete + a few releases to give people time to upgrade.

Reply:
Do you have a suggestion or workaround? We need to wait for elasticsearch to start to insert the securityconfig.

Reply:
Here's an example of Builder doing this: https://github.com/habitat-sh/builder/blob/master/components/builder-minio/habitat/config/hook-helper.sh#L70

I'm not happy about it either, but we need some method of waiting for the service to be available. Can't we be practical here?

Reply:
Reading through this again, I missed the BREAKER bits on the first pass. That should be sufficient to prevent a permanently deadlocked supervisor, though it will still appear hung during the execution.  I'd ask for two small changes though:

A comment above the while loop explaining that because this runs in the supervisors main thread, the supervisor will be unresponsive while this is executing and sleeps are generally not a good practice to rely on in lifecycle hooks.

Bring the breaker total time to expected time to start on a reasonable system +~20-50%. It's at 5 minutes right now which seems rather long, but I don't have the context on how long elastic takes to start up.

Reply:
Done and done! I agree that 5m is maybe too long, so I brought it down to 2.5 minutes.

I also added a comment with the relevant bits. I also did some experiments with curl doing a retry / backoff sleep, but this does effectively the same thing (and it's less obvious, so it makes it harder to refactor or fix in the future).